src/algebra/polynomial/big_operators.lean

```diff
@@ -277,7 +277,18 @@ by simpa using multiset_prod_X_sub_C_coeff_card_pred (s.1.map f) (by simpa using
 end comm_ring
 
 section no_zero_divisors
-variables [comm_ring R] [no_zero_divisors R] (f : ι → R[X]) (t : multiset R[X])
+
+section semiring
+variables [semiring R] [no_zero_divisors R]
+
+lemma degree_list_prod [nontrivial R] (l : list R[X]) :
```

Comment:
Could you please add a docstring explaining that `[nontrivial R]` is needed because otherwise for `l = []` we have `bot` in the LHS and `0` in the RHS?
Otherwise LGTM
bors d+